### PR TITLE
Added spec location verification to the release pipeline

### PR DESCRIPTION
**Pull request by raych1**
## Purpose of this PR
The purpose is to gate on the REST API specifications used to release SDK are from the main branch of Azure/azure-rest-api-specs repository.

## Linked PRs
The common script and pipeline are merged by another PR.
https://github.com/Azure/azure-sdk-tools/pull/7451

FYI - The same change has been added to the other four tier 1 language release pipelines.
https://github.com/Azure/azure-sdk-for-net/pull/40783

Because Go SDK doesn't have package generated, it needs to add a step to publish the `PackageInfo` as artifact. `PackageInfo` will be used in downstream step to get the sdk directory and the version info.

This PR should hold on the merge until below PR is merged.
https://github.com/Azure/azure-sdk-tools/pull/7585

## Test PR
[aztemplate test pipeline run](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3441736&view=logs&j=6a8552c2-2ada-5ee5-a4d2-6fcf45f87c55)

@weshaggard @benbp , can you please review this PR?

### CODE REVIEW

File: eng/pipelines/templates/steps/analyze.yml [hunk 1]
```diff
@@ -18,7 +18,10 @@ steps:
       workingDirectory: $(Pipeline.Workspace)
     displayName: Dump Package properties
     condition: and(succeeded(), ${{ parameters.IsSdkLibrary }})
-
+  - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:21]: @benbp is the analyze job the best place for this? Should it instead go in the Build job? 

@raych1 if this stays here then what happens for anyone that sets isSdkLibrary false? they won't have any PackgeInfo files produced.

Comment (weshaggard) [reply eng/pipelines/templates/steps/analyze.yml:21]: If IsSdkLibrary is set to false then this publish artifact step will fail as there will not be any files to publish. We might need to add the same condition on the publish step.

**Comment by raych1** [comment]
@weshaggard ,

Here is the [latest test run](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3450488&view=logs&j=6a8552c2-2ada-5ee5-a4d2-6fcf45f87c55&t=63abb2ae-8b99-518c-37b5-ceb4c0582ab3) which works as expected. 

This is [another test run](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3450726&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=bdeefc16-b669-5ebd-ad94-a2c19ade53b0) for a tooling pipeline instead of SDK release which also works as expected.

File: eng/pipelines/templates/steps/analyze.yml (hunk 1 above)
Comment (raych1) [reply eng/pipelines/templates/steps/analyze.yml]: Good catch, I added the same condition.

**Review by weshaggard — APPROVED** [review]

**Comment by raych1** [comment]
@jhendrixMSFT , this PR will impact the data plane SDK release. Let me know if you have any concerns.

**Pull request merged**